File: mycollab-services/src/main/scala/com/mycollab/module/project/domain/criteria/ProjectGenericItemSearchCriteria.scala
package com.mycollab.module.project.domain.criteria

import com.mycollab.db.arguments.{SearchCriteria, SetSearchField, StringSearchField}

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
class ProjectGenericItemSearchCriteria extends SearchCriteria {
  @BeanProperty var prjKeys: SetSearchField[Integer] = _
  @BeanProperty var txtValue: StringSearchField = _
  @BeanProperty var createdUsers: SetSearchField[String] = _
  @BeanProperty var types: SetSearchField[String] = _
  @BeanProperty var monitorProjectIds: SetSearchField[Integer] = _
  @BeanProperty var tagNames: SetSearchField[String] = null
}

File: mycollab-services/src/main/java/com/mycollab/module/project/domain/criteria/ProjectGenericItemSearchCriteria.kt
package com.mycollab.module.project.domain.criteria

import com.mycollab.db.arguments.SearchCriteria
import com.mycollab.db.arguments.SetSearchField
import com.mycollab.db.arguments.StringSearchField

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class ProjectGenericItemSearchCriteria : SearchCriteria(){
    lateinit var prjKeys: SetSearchField<Int>
    lateinit var txtValue: StringSearchField
    lateinit var createdUsers: SetSearchField<String> 
    lateinit var types: SetSearchField<String> 
    lateinit var monitorProjectIds: SetSearchField<Int> 
    lateinit var tagNames: SetSearchField<String> 
}